go/mysql/flavor_mysql_legacy.go

```diff
@@ -0,0 +1,278 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mysql
+
+import (
+	"fmt"
+	"strings"
+
+	"vitess.io/vitess/go/mysql/replication"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+// mysqlFlavor implements the Flavor interface for Mysql.
+type mysqlFlavorLegacy struct {
+	mysqlFlavor
+}
+type mysqlFlavor57 struct {
+	mysqlFlavorLegacy
+}
+type mysqlFlavor8Legacy struct {
+	mysqlFlavorLegacy
+}
```

Comment:
So many flavours of MySQL is now getting confusing especially because they're inheriting methods too. How are `mysqlFlavor`, `mysqlFlavorLegacy` different? Previously `mysqlFlavor` had the base implementations for all commands and overrides were there in other specific implementations, but now there is `mysqlFlavorLegacy` which is used in mysqlFlavor57 and mysqlFlavor8Legacy. There is also mysqlFlavor8 and mysqlFlavor82...

Maybe its just me but this new structure of so many flavors is confusing to me.

Reply:
@GuptaManan100 So the idea here is that `mysqlFlavor` is the latest version of what is present always. `mysqlLegacyFlavor` overrides then all the cases for MySQL version before any of the changes were available. 

The annoying part here is that the new syntax was added incrementally in MySQL, which means that even for MySQL 8.0.x version that have deprecated the `slave` terminology, they don't have all the changes to deprecate `master`. 

Only starting with 8.2.0 and later is all the terminology available. I opted for this setup to minimize the amount of things that needed to be overridden / copied. I'm not sure if there's really a much simpler way to be honest at this point.

Reply:
Okay, it would be great if we can document what each of the flavours represents, especially `mysqlFlavour` and `mysqlLegacyFlavour`